src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java

```diff
@@ -0,0 +1,1406 @@
+import purejavacomm.CommPort;
+import purejavacomm.CommPortIdentifier;
+import purejavacomm.NoSuchPortException;
+import purejavacomm.PortInUseException;
+import purejavacomm.PureJavaIllegalStateException;
+import purejavacomm.SerialPort;
+import purejavacomm.UnsupportedCommOperationException;
+
+public class Ruida extends LaserCutter
+{
+  private static final int MINFOCUS = -500; //Minimal focus value (not mm)
+  private static final int MAXFOCUS = 500; //Maximal focus value (not mm)
+  private static final int MAXPOWER = 80;
+  private static final double FOCUSWIDTH = 0.0252; //How much mm/unit the focus values are
+  protected static final String SETTING_HOST = "IP/Hostname";
+  protected static final String SETTING_COMPORT = "USB device";
+  protected static final String SETTING_SERIAL_TIMEOUT = "Milliseconds to wait for response";
+  protected static final String SETTING_FILE_EXPORT_PATH = "Path to save exported code";
+  protected static final String SETTING_UPLOAD_METHOD = "Upload method";
+  protected static final String UPLOAD_METHOD_FILE = "File";
+  protected static final String UPLOAD_METHOD_IP = "IP";
+  protected static final String UPLOAD_METHOD_SERIAL = "USB";
+
+  protected static final String SETTING_MAX_VECTOR_CUT_SPEED = "Max vector cutting speed (mm/s)";
+  protected static final String SETTING_MAX_VECTOR_MOVE_SPEED = "Max vector move speed (mm/s)";
+  protected static final String SETTING_MIN_POWER = "Min laser power (%)";
+  protected static final String SETTING_MAX_POWER = "Max laser power (%)";
+  protected static final String SETTING_BED_WIDTH = "Bed width (mm)";
+  protected static final String SETTING_BED_HEIGHT = "Bed height (mm)";
+  protected static final String SETTING_USE_BIDIRECTIONAL_RASTERING = "Use bidirectional rastering";
+  protected static final Locale FORMAT_LOCALE = Locale.US;
+  // config values
+  private static final long[] JogAcceleration = {200000,50000,600000};
+  private static final long[] JogMaxVelocity = {16,16,2048};
+  private static final long[] EngraveAcceleration = {200000,50000,600000};
+  private static final long[] EngraveMaxVelocity = {800,800,2048};
+  private static final long[] VectorAcceleration = {100000,25000,20000};
+  private static final long[] VectorMaxVelocity = {1000,1000,1000};
+//  private static final byte FlipLaserPWMPower = 1;
+//  private static final byte FlipLaserOutput = 0;
+
+//  private static final byte HomeDirection = 1;
+//  private static final byte[] FlipHomeDirection = {1,0,0};
+//  private static final byte[] LimitContCondition = {0,0,0,0};
+//  private static final long[] MaxSteps = {250,500,500};
+//  private static final long[] TableSize = {20000,12000,30000};
+
+  protected static final String[] uploadMethodList = {UPLOAD_METHOD_FILE, UPLOAD_METHOD_IP, UPLOAD_METHOD_SERIAL};
+  public static final int SOURCE_PORT = 40200; // used by rdworks in Windows
+  public static final int DEST_PORT = 50200; // fixed UDP port
+
+  private ByteStream stream;
```

Comment:
should be transient to hide it from the serialization?

Reply:
moved into `UdpStream`

Reply:
Huh ?
`SOURCE_PORT` and `DEST_PORT` are in `class UdpStream` now and (esp.) don't appear in `~/.visicut/devices/Ruida.xml` any more.
Am I missing anything ?

Reply:
I was referring to `stream`, which should be `transient` so that it is definitely not saved to the XML file.

Reply:
fixed